File: boat-spring-boot-sample/src/test/java/sample/java/xyz/srclab/spring/boot/bean/MyBeanRegistry.java
package sample.java.xyz.srclab.spring.boot.bean;

import org.jetbrains.annotations.NotNull;
import xyz.srclab.spring.boot.bean.BeanProperties;
import xyz.srclab.spring.boot.bean.BeanRegistry;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class MyBeanRegistry extends BeanRegistry {

    @NotNull
    @Override
    protected Map<String, Object> registeredSingletons() {
        Map<String, Object> result = new HashMap<>();
        result.put("bean1", "bean1");
        result.put("bean2", "bean2");
        return result;
    }

    @NotNull
    @Override
    protected Set<BeanProperties> registeredBeans() {
        Set<BeanProperties> result = new HashSet<>();
        BeanProperties beanProperties = new BeanProperties();
        beanProperties.setName("myBean");
        beanProperties.setClassName(MyBean.class.getName());
        result.add(beanProperties);
        return result;
    }
}

File: boat-spring-boot-sample/src/test/kotlin/sample/kotlin/xyz/srclab/spring/boot/bean/MyBeanRegistry.kt
package sample.kotlin.xyz.srclab.spring.boot.bean

import org.springframework.stereotype.Component
import xyz.srclab.spring.boot.bean.BeanProperties
import xyz.srclab.spring.boot.bean.BeanRegistry

@Component
open class MyBeanRegistry : BeanRegistry() {

    override val registeredSingletons: Map<String, Any>
        get() {
            val result: MutableMap<String, Any> = HashMap()
            result["bean1"] = "bean1"
            result["bean2"] = "bean2"
            return result
        }

    override val registeredBeans: Set<BeanProperties>
        get() {
            val result: MutableSet<BeanProperties> = HashSet()
            val beanProperties = BeanProperties()
            beanProperties.name = "myBean"
            beanProperties.className = MyBean::class.java.name
            result.add(beanProperties)
            return result
        }
}